src/main/java/org/cryptimeleon/mclwrap/bn254/MclBilinearGroupImpl.java

```diff
@@ -44,19 +49,96 @@ public static boolean isAvailable() {
         return isInitialized;
     }
 
+    protected static boolean loadIncludedLibrary(){
+        final String platformName = System.getProperty("os.name").toLowerCase();
+        final String platformArch = System.getProperty("os.arch").toLowerCase();
+
+        String requiredLibraryName = null;
+
+        // for Windows, OS and arch names defined in windows/native/libjava/java_props_md.c
+        // should be stable
+        if(platformName.contains("win")){
+            if(platformArch.equals("x86")){
+                requiredLibraryName="mcljava-win-x86.dll";
+            }
+            if(platformArch.equals("amd64")){
+                requiredLibraryName="mcljava-win-x64.dll";
+            }
+        }
+        // OS name on Linux / BSDs is as returned by uname(2).
+        // it is unclear whether the naming on linux is stable when it comes to Intel 32 bit
+        if(platformName.contains("linux")){
+            if(platformArch.equals("x86") || platformArch.equals("i386")){
+                requiredLibraryName="mcljava-linux-x86.so";
+            }
+            if(platformArch.equals("amd64")){
+                requiredLibraryName="mcljava-linux-x64.so";
+            }
+        }
+
+        if(platformName.contains("mac")){
+            if(platformArch.equals("amd64")){
+                requiredLibraryName="mcljava-mac-x64.dylib";
+            }
+        }
+
+        if(requiredLibraryName == null){
+            return false;
+        }
+        InputStream nativeLibrary = MclBilinearGroupImpl.class.getResourceAsStream(requiredLibraryName);
+
+        if(nativeLibrary == null){
+            return false;
+        }
+
+        String tempdir;
+        try {
+            tempdir = Files.createTempDirectory("mclwrap").toAbsolutePath().toString();
+        } catch (IOException e) {
+            return false;
+        }
+
+        File outputFile = new File(tempdir+File.separator+requiredLibraryName);
+
+        if(!outputFile.exists()){
```

Comment:
I'd vote to replace this with `if(true)`, essentially. I don't see the user providing a library by writing it into a tmp dir.
On the flip side, this if may have weird consequences when, for example, we upgrade mcl binaries at some point and someone still has the old ones in the tmp dir.

Reply:
Done.